### PR TITLE
Service Fabric - add managedcluster and nodetype specs

### PR DESCRIPTION
**Pull request by a-santamaria**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/servicefabric/resource-manager/Microsoft.ServiceFabric/preview/2020-01-01-preview/managedcluster.json [hunk 1]
```diff
@@ -0,0 +1,1080 @@
...
+            "schema": {
+              "$ref": "#/definitions/ErrorModel"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.ServiceFabric/managedClusters": {
+      "get": {
+        "operationId": "ManagedClusters_List",
```
Comment (sanjaiganesh) [inline specification/servicefabric/resource-manager/Microsoft.ServiceFabric/preview/2020-01-01-preview/managedcluster.json]: >ManagedClusters_List [](start = 24, length = 20)

_ListBySubscription

File: specification/servicefabric/resource-manager/Microsoft.ServiceFabric/preview/2020-01-01-preview/managedcluster.json [hunk 2]
```diff
@@ -0,0 +1,1080 @@
...
+              "$ref": "#/definitions/ErrorModel"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "ManagedCluster"
+        ],
+        "operationId": "ManagedClusters_Create",
```
Comment (sanjaiganesh) [inline specification/servicefabric/resource-manager/Microsoft.ServiceFabric/preview/2020-01-01-preview/managedcluster.json]: >ManagedClusters_Create [](start = 24, length = 22)

CreateOrUpdate

File: specification/servicefabric/resource-manager/Microsoft.ServiceFabric/preview/2020-01-01-preview/managedcluster.json [hunk 3]
```diff
@@ -0,0 +1,1080 @@
...
+        ],
+        "x-ms-examples": {
+          "Put a cluster with minimum parameters": {
+            "$ref": "./examples/ManagedClusterPutOperation_example_min.json"
+          },
+          "Put a cluster with maximum parameters": {
+            "$ref": "./examples/ManagedClusterPutOperation_example_max.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
```
Comment (sanjaiganesh) [inline specification/servicefabric/resource-manager/Microsoft.ServiceFabric/preview/2020-01-01-preview/managedcluster.json]: >"x-ms-long-running-operation": true, [](start = 8, length = 36)

also add "x-ms-long-running-operation-options" extension with "final-state-via" property.
``` json
"x-ms-long-running-operation-options": {
  "final-state-via": "location"
}
```

or

``` json
"x-ms-long-running-operation-options": {
  "final-state-via": "azure-async-operation"
}
```

Comment (sanjaiganesh) [reply specification/servicefabric/resource-manager/Microsoft.ServiceFabric/preview/2020-01-01-preview/managedcluster.json]: same for all LRO operations (with 201 or 202)

---
In reply to: [368180555](https://github.com/Azure/azure-rest-api-specs/pull/8003#discussion_r368180555) [](ancestors = 368180555)

File: specification/servicefabric/resource-manager/Microsoft.ServiceFabric/preview/2020-01-01-preview/managedcluster.json [hunk 4]
```diff
@@ -0,0 +1,1080 @@
...
+          {
+            "$ref": "#/parameters/subscriptionId"
+          }
+        ],
+        "x-ms-examples": {
+          "Delete a cluster": {
+            "$ref": "./examples/ManagedClusterDeleteOperation_example.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
```
Comment (sanjaiganesh) [inline specification/servicefabric/resource-manager/Microsoft.ServiceFabric/preview/2020-01-01-preview/managedcluster.json]: >x-ms-long-running-operation": true, [](start = 9, length = 35)

also add "x-ms-long-running-operation-options" extension with "final-state-via" property.
``` json
"x-ms-long-running-operation-options": {
  "final-state-via": "location"
}
```

or

``` json
"x-ms-long-running-operation-options": {
  "final-state-via": "azure-async-operation"
}
```

File: specification/servicefabric/resource-manager/Microsoft.ServiceFabric/preview/2020-01-01-preview/managedcluster.json [hunk 5]
```diff
@@ -0,0 +1,1080 @@
...
+            "in": "body",
+            "description": "The parameters to update the managed cluster configuration.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ManagedClusterUpdateParameters"
+            },
+            "x-ms-parameter-location": "method"
+          }
+        ],
+        "x-ms-long-running-operation": true,
```
Comment (sanjaiganesh) [inline specification/servicefabric/resource-manager/Microsoft.ServiceFabric/preview/2020-01-01-preview/managedcluster.json]: > "x-ms-long-running-operation": true, [](start = 7, length = 37)

also add "x-ms-long-running-operation-options" extension with "final-state-via" property.
``` json
"x-ms-long-running-operation-options": {
  "final-state-via": "location"
}
```

or

``` json
"x-ms-long-running-operation-options": {
  "final-state-via": "azure-async-operation"
}
```

File: specification/servicefabric/resource-manager/Microsoft.ServiceFabric/preview/2020-01-01-preview/nodetype.json [hunk 6]
```diff
@@ -0,0 +1,815 @@
...
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/Microsoft.ServiceFabric/managedClusters/{clusterName}/nodeTypes": {
+      "get": {
+        "operationId": "NodeTypes_List",
```
Comment (sanjaiganesh) [inline specification/servicefabric/resource-manager/Microsoft.ServiceFabric/preview/2020-01-01-preview/nodetype.json]: >NodeTypes_List [](start = 24, length = 14)

_ListByManagedClusters

File: specification/servicefabric/resource-manager/Microsoft.ServiceFabric/preview/2020-01-01-preview/nodetype.json [hunk 7]
```diff
@@ -0,0 +1,815 @@
...
+        ],
+        "x-ms-examples": {
+          "Put a node type with minimum parameters": {
+            "$ref": "./examples/NodeTypePutOperation_example_min.json"
+          },
+          "Put a node type with maximum parameters": {
+            "$ref": "./examples/NodeTypePutOperation_example_max.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
```
Comment (sanjaiganesh) [inline specification/servicefabric/resource-manager/Microsoft.ServiceFabric/preview/2020-01-01-preview/nodetype.json]: > "x-ms-long- [](start = 7, length = 12)

also  add "x-ms-long-running-operation-options" extension with "final-state-via" property.
``` json
"x-ms-long-running-operation-options": {
  "final-state-via": "location"
}
```

or

``` json
"x-ms-long-running-operation-options": {
  "final-state-via": "azure-async-operation"
}
```

File: specification/servicefabric/resource-manager/Microsoft.ServiceFabric/preview/2020-01-01-preview/nodetype.json [hunk 8]
```diff
@@ -0,0 +1,815 @@
...
+            "in": "body",
+            "description": "The parameters to update the node type configuration.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/NodeTypeUpdateParameters"
+            },
+            "x-ms-parameter-location": "method"
+          }
+        ],
+        "x-ms-long-running-operation": true,
```
Comment (sanjaiganesh) [inline specification/servicefabric/resource-manager/Microsoft.ServiceFabric/preview/2020-01-01-preview/nodetype.json]: > "x-ms-long-runn [](start = 7, length = 16)

 add "x-ms-long-running-operation-options" extension with "final-state-via" property.
``` json
"x-ms-long-running-operation-options": {
  "final-state-via": "location"
}
```

or

``` json
"x-ms-long-running-operation-options": {
  "final-state-via": "azure-async-operation"
}
```

File: specification/servicefabric/resource-manager/Microsoft.ServiceFabric/preview/2020-01-01-preview/nodetype.json [hunk 9]
```diff
@@ -0,0 +1,815 @@
...
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceFabric/managedClusters/{clusterName}/nodeTypes/{nodeTypeName}": {
+      "get": {
+        "tags": [
+          "NodeType"
+        ],
+        "operationId": "NodeTypes_Get",
```
Comment (sanjaiganesh) [inline specification/servicefabric/resource-manager/Microsoft.ServiceFabric/preview/2020-01-01-preview/nodetype.json]: >NodeTypes_Get [](start = 24, length = 13)

add _ListByResourceGroup and _ListByScubscription apis as well

**Comment by sanjaiganesh** [comment]
@a-santamaria added some feedback.   If you happen to resolve next week, follow up with Marcin (oncall next week) for approval on ARM side.

**Comment by akning-ms** [comment]
Hi @a-santamaria any reply for Sanjai's comments as it is pending for long time?

**Comment by ravbhatnagar** [comment]
@akning-ms - is there a label we can use to indicate old PRs where we are waiting PR requestors reply? Such RPs cause a lot of noise to the reviewers. Or may be we close it. If it needs to be re-opened, the PR requestor can ping. Thoughts?

**Comment by a-santamaria** [comment]
thanks, we had a review with ARM and only made it available in our test region while we develop it. I'll close it for now and will reopen or create a new one once we are ready for the preview.

**Pull request closed without merging**